### DISCUSSION TITLE
Fish + Kitty repaint visual bug

### DISCUSSION THREAD

**Discussion by iacore** [fish-shell/fish-shell#10289] [Q&A] [answered]
Hello, fish users. I have found a strange bug when using fish with [kitty](https://github.com/kovidgoyal/kitty/) (terminal).

I am not sure what problem is this, or is it a bug at all, so I'm sharing it here.

## How to reproduce

First, open fish in kitty.

Run this:

```fish
    bind \eg 'echo; git status; git remote -v; echo; commandline -f repaint'
```

In a git repo, press <kbd>Alt+g</kbd>.

Then resize the terminal window. (not sure how to produce it consistently)

## What I see

The text printed by `git status; git remote -v` disappears!

## What I expect

The text should stay in the buffer.

## Extra Information

I guess the output of the command ran by hotkey is not committed to the scrollback buffer?

**Comment by faho** [comment]
Works in my terminal (which is not kitty).

> I guess the output of the command ran by hotkey is not committed to the scrollback buffer?

Fish has no influence on what is committed to which buffer.

It prints text, and moves the cursor around. In particular `commandline -f repaint` will tell fish to run and draw the prompt again on the line it is currently on. Since you used `echo`, that will be *after* the `git remote` output.

I would guess that kitty is too smart here, and e.g. adds integration that marks the prompt.

**Reply by iacore** [reply]
thanks

**Comment by iacore** [comment]
I think I found the reason why this is happening. See https://github.com/kovidgoyal/kitty/issues/7200#issuecomment-1980794267.

If the `fish_prompt` event is fired before a new prompt is shown, should it also be fired after `commandline -f repaint`? (currently no)

**Reply by faho** [reply, marked as answer]
Fish now writes the OSC 133 prompt markers - https://github.com/fish-shell/fish-shell/commit/3b9e3e251bf9d4c7d0b31275cac55df68fe0127a.

That bit in kitty's integration is therefore obsolete in the upcoming fish 4.0.
